Lib/subprocess.py

```diff
@@ -815,6 +836,13 @@ def __init__(self, args, bufsize=-1, executable=None,
         self.encoding = encoding
         self.errors = errors
         self.pipesize = pipesize
+        if read_cb is not None:
+            if not callable(read_cb):
+                ValueError("read_cb not a callable")
+            self.read_cb = read_cb
+        else:
+            self.read_cb = Popen._read_handler
```

Comment:
> it doesn't need self, per my other comment just pass an indicator of if the call is being made for stdout or for stderr data. Or have a separate read_callback parameter for each instead so that conditional logic need not be duplicated within callback implementations.

If someone derives from `Popen` and adds additional internal state, we can't know what the handler will need access to and what it won't.  The only way around this is giving the handler access to `self`.